### ISSUE TITLE
Sliced Iterator similar to Scala Iterator.slice method

### ISSUE THREAD

**Issue by fanifieiev** [yegor256/cactoos#1188]
I could not find any implementation of sliced iterator, so I think it would be nice to have it in place.
Constructor:
`public Sliced(final int fromIndex, final Iterator<T> iterator, final int limit)`
An example is:
`Iterator<Integer> sliced = new Sliced<>(2, new IteratorOf<>(1,2,3,4,5,6), 2);`
that would return a slice of 2 elements starting from index 2, which will result in [3, 4].

PS: I have already implemented one and would like to create PR after this issue is created.

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) please, pay attention to this issue

**Comment by 0crat** [comment]
@fanifieiev[/z](https://www.0crat.com/u/fanifieiev) this project will fix the problem faster if you donate a few dollars to it; just [click here](https://www.0crat.com/contrib/C63314D6Z) and pay via Stripe, it's very fast, convenient and appreciated; thanks a lot!

**Comment by paulodamaso** [comment]
@fanifieiev I'd like it, BUT:
We have some other similar behavior in cactoos project and I think that we can reuse the code between them:
`HeadOf`: gets the first N elements of the collection
`TailOf`: gets the last N elements of the collection
`Skipped`: skip N elements of the collection
`Partitioned`: slices the collection in parts of equal size N

If we look carefully we could compose these decorators using the new proposed `Sliced` one:
- `HeadOf` is an slice from 0 to N
- `TailOf` is an slice from size-N to size 
- `Skipped` is an slice from N to size
- `Partitioned` is a collection of slices of size/N, starting at 0 to N, N+1 to 2N, and so on

So let's do it: create the `Sliced` decorator and use it in the aforementioned implementation.

P.S.: Let's create it for all interfaces: Collection, Iterable, Iterator, etc

**Comment by paulodamaso** [comment]
@0crat in

**Comment by paulodamaso** [comment]
@0crat assign @fanifieiev

**Comment by 0crat** [comment]
> @0crat in ([here](https://github.com/yegor256/cactoos/issues/1188#issuecomment-532788325))

@paulodamaso Job [#1188](https://github.com/yegor256/cactoos/issues/1188) is already in scope

**Comment by 0crat** [comment]
> @0crat assign @fanifieiev  ([here](https://github.com/yegor256/cactoos/issues/1188#issuecomment-532788449))

@paulodamaso Thanks for your contribution, @fanifieiev[/z](https://www.0crat.com/u/fanifieiev)! If you would be a member of [the project](https://www.0crat.com/p/C63314D6Z), you would now earn +15 reputation points, as explained in [§29](http://www.zerocracy.com/policy.html#29). You can join and apply to it, see [§2](http://www.zerocracy.com/policy.html#2).

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
It is strongly discouraged to assign jobs to their creators, see [§19](http://www.zerocracy.com/policy.html#19): -15 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by fabriciofx** [comment]
@fanifieiev @paulodamaso II'm sorry, bu I didn't like this API. WDYT something like:
```
public Sliced(final int start, final int count, final T items...) {}
public Sliced(final int start, final int count, final Iterable<T> iterable) {}
```

**Comment by fanifieiev** [comment]
> @fanifieiev @paulodamaso II'm sorry, bu I didn't like this API. WDYT something like:
> 
> ```
> public Sliced(final int start, final int count, final T items...) {}
> public Sliced(final int start, final int count, final Iterable<T> iterable) {}
> ```

@fabriciofx Is your concern related to the constructor parameter names?

**Comment by fanifieiev** [comment]
@0crat review

**Comment by paulodamaso** [comment]
@fabriciofx What's the problem about that?

**Comment by fanifieiev** [comment]
> @fabriciofx What's the problem about that?

@paulodamaso I have already refactored the api, the changes are in there, in PR

**Comment by fabriciofx** [comment]
@paulodamaso It didn't follow the Cactoos API pattern and IMHO it's more clear now. Ah, in time: I'm not the `REV` of this PR...

**Comment by 0crat** [comment]
> @0crat review [#1189](https://github.com/yegor256/cactoos/pull/1189) ([here](https://github.com/yegor256/cactoos/issues/1188#issuecomment-532972948))

@fanifieiev Can't understand "review", try one of these:
  - `assign` Assign a performer to this issue
  - `boost` Set a boost for the job
  - `continue` Remove a job's impediment
  - `hello` Just say hello
  - `in` Register this issue in scope (WBS)
  - `out` Close the order and remove this job from scope
  - `pay` Pay a user some extra cash
  - `quality` Review a task
  - `resign` Resign from this issue
  - `status` Check the status of the job
  - `version` Print current version of the bot
  - `wait` Register an impediment for a job

**Comment by fabriciofx** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/cactoos/issues/1188#issuecomment-533280670))

@fabriciofx This is what I know about this job in [C63314D6Z](https://www.0crat.com/p/C63314D6Z), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#1188](https://github.com/yegor256/cactoos/issues/1188) is [in scope](http://www.zerocracy.com/policy.html#14) for 1days
  * The role is `DEV`
  * The job is assigned to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev) for [1days](http://www.zerocracy.com/policy.html#8)
  * There is no monetary reward attached, it's a [free](http://www.zerocracy.com/policy.html#2) job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 30 minutes/points
  * These users are banned and won't be assigned:
    * @fanifieiev[/z](https://www.0crat.com/u/fanifieiev): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C63314D6Z?q=%7Bjob%3A%27gh%3Ayegor256%2Fcactoos%231188%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by fanifieiev** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/cactoos/issues/1188#issuecomment-533313273))

@fanifieiev This is what I know about this job in [C63314D6Z](https://www.0crat.com/p/C63314D6Z), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#1188](https://github.com/yegor256/cactoos/issues/1188) is [in scope](http://www.zerocracy.com/policy.html#14) for 1days
  * The role is `DEV`
  * The job is assigned to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev) for [1days](http://www.zerocracy.com/policy.html#8)
  * There is no monetary reward attached, it's a [free](http://www.zerocracy.com/policy.html#2) job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 30 minutes/points
  * These users are banned and won't be assigned:
    * @fanifieiev[/z](https://www.0crat.com/u/fanifieiev): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C63314D6Z?q=%7Bjob%3A%27gh%3Ayegor256%2Fcactoos%231188%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by fanifieiev** [comment]
@paulodamaso 
Why am I banned? 
The Contribution section of the page [Cactoos](https://github.com/yegor256/cactoos) says:

> How to contribute?
Just fork the repo and send us a pull request.
Make sure your branch builds without any warnings/issues:

**Comment by paulodamaso** [comment]
@fanifieiev You're banned from DEV role in the issue because you have opened it. Zerocrat does not assign issues to its creators. Your changes are being evaluated by @victornoel in #1189

**Comment by fanifieiev** [comment]
> @fanifieiev You're banned from DEV role in the issue because you have opened it. Zerocrat does not assign issues to its creators. Your changes are being evaluated by @victornoel in #1189
@paulodamaso 
Ok, got it, I created the issue, but the assignment was done on behalf of you, right?
So, what is the proper way to join DEV ?
Thanks

**Comment by paulodamaso** [comment]
@fanifieiev You need to join zerocracy in order to get reputation points and to be paid for jobs that have monetary rewards. Please read the [policy](http://www.zerocracy.com/policy.html)

**Comment by 0pdd** [comment]
@fanifieiev 4 puzzles [#1191](https://github.com/yegor256/cactoos/issues/1191), [#1192](https://github.com/yegor256/cactoos/issues/1192), [#1193](https://github.com/yegor256/cactoos/issues/1193), [#1194](https://github.com/yegor256/cactoos/issues/1194) are still not solved.

**Comment by 0pdd** [comment]
@fanifieiev 3 puzzles [#1191](https://github.com/yegor256/cactoos/issues/1191), [#1193](https://github.com/yegor256/cactoos/issues/1193), [#1194](https://github.com/yegor256/cactoos/issues/1194) are still not solved; solved: [#1192](https://github.com/yegor256/cactoos/issues/1192).

**Comment by victornoel** [comment]
@fanifieiev this should be closed now, the tickets created by 0pdd will take over now for the rest of the work.

**Comment by fanifieiev** [comment]
> @fanifieiev this should be closed now, the tickets created by 0pdd will take over now for the rest of the work.

Great

**Comment by victornoel** [comment]
@fanifieiev yes, that's what I meant :)

**Comment by fanifieiev** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/cactoos/issues/1188#issuecomment-541159451))

@fanifieiev This is what I know about this job in [C63314D6Z](https://www.0crat.com/p/C63314D6Z), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#1188](https://github.com/yegor256/cactoos/issues/1188) is [in scope](http://www.zerocracy.com/policy.html#14) for 23days
  * The role is `DEV`
  * The job is assigned to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev) for [23days](http://www.zerocracy.com/policy.html#8)
  * There is no monetary reward attached, it's a [free](http://www.zerocracy.com/policy.html#2) job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 30 minutes/points
  * These users are banned and won't be assigned:
    * @fanifieiev[/z](https://www.0crat.com/u/fanifieiev): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C63314D6Z?q=%7Bjob%3A%27gh%3Ayegor256%2Fcactoos%231188%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by fanifieiev** [comment]
Work is done!!!

**Comment by 0crat** [comment]
Job [#1188](https://github.com/yegor256/cactoos/issues/1188) is not in the agenda of @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), can't retrieve data and time of add

**Comment by 0crat** [comment]
Job [#1188](https://github.com/yegor256/cactoos/issues/1188) is not in the agenda of @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), can't inspect

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1188](https://github.com/yegor256/cactoos/issues/1188) is now out of scope

**Comment by sereshqua** [comment]
@fanifieiev 

1) we need to make sure that problem reporter and problem solver are two different persons;
2) you start all the comments from the name of the person you are referring to, [see](https://github.com/yegor256/cactoos/issues/1188#issuecomment-541184305)

**Comment by sereshqua** [comment]
@fanifieiev ping

**Comment by fanifieiev** [comment]
@sereshqua 
1. I was the problem reporter and problem solver at the same time, but I was not part of this project at that time. When I created the problem, I was assigned by @paulodamaso to solve it. I did not know the rules then.

2. How should the problem be closed ?
    I was asked by reviewer @victornoel to close it manually, [see](https://github.com/yegor256/cactoos/issues/1188#issuecomment-541040708)

**Comment by sereshqua** [comment]
@fanifieiev 

1) it is a policy rule and 0crat shows that the job has been completed by you, [see](https://github.com/yegor256/cactoos/issues/1188#issuecomment-541195284)

2) you need to start EVERY single comment in both issues & PRs with the name of the person your referring to.

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +30 point(s) just awarded to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by fanifieiev** [comment]
> @fanifieiev
> 
> 1. it is a policy rule and 0crat shows that the job has been completed by you, [see](https://github.com/yegor256/cactoos/issues/1188#issuecomment-541195284)
> 2. you need to start EVERY single comment in both issues & PRs with the name of the person your referring to.

@sereshqua Got it, will do next time.

**Comment by 0pdd** [comment]
@fanifieiev the puzzle [#1194](https://github.com/yegor256/cactoos/issues/1194) is still not solved; solved: [#1191](https://github.com/yegor256/cactoos/issues/1191), [#1192](https://github.com/yegor256/cactoos/issues/1192), [#1193](https://github.com/yegor256/cactoos/issues/1193).

**Comment by 0pdd** [comment]
@fanifieiev all 4 puzzles are solved here: [#1191](https://github.com/yegor256/cactoos/issues/1191), [#1192](https://github.com/yegor256/cactoos/issues/1192), [#1193](https://github.com/yegor256/cactoos/issues/1193), [#1194](https://github.com/yegor256/cactoos/issues/1194).
